pkg/prometheus/operator.go

```diff
@@ -853,13 +853,77 @@ func (c *Operator) destroyPrometheus(key string) error {
 	return nil
 }
 
+func (c *Operator) loadBasicAuthSecrets(mons map[string]*v1alpha1.ServiceMonitor, s *v1.SecretList) (map[string]interface{}, error) {
+
+	secrets := map[string]interface{}{}
```

Comment:
We're sort of mixing the config generation and secret loading part here. As long as we can I think it would be best to stay type-safe here and leave the `interface{}{}` doings to completely up to the config generator. I'd suggest to introduce a new struct here, like

```
type basicAuthCredentials struct {
  username string
  password string
}
```

And use that instead of `map[string]interface{}{}`, and then in the config generation step convert that to `map[string]interface{}{}`. That way we can rely on the fields existing when we pass it to the config generator.